src/pal/src/exception/signal.cpp

```diff
@@ -697,6 +697,42 @@ PAL_ERROR InjectActivationInternal(CorUnix::CPalThread* pThread)
 
 /*++
 Function :
+    signal_ignore_handler
+
+    Simple signal handler which does nothing
+
+Parameters :
+    POSIX signal handler parameter list ("man sigaction" for details)
+
+(no return value)
+--*/
+static void signal_ignore_handler(int code, siginfo_t *siginfo, void *context)
+{
+}
+
+
+void PAL_IgnoreProfileSignal(void)
+{
+#if !HAVE_MACH_EXCEPTIONS && defined(SIGRTMAX)
```

Comment:
What if rather than having the SIGRTMAX hardcoded in the code, we would add a new config option  (COMPlus_XXXXX) that would contain the number of signal that we want to ignore? There are no realtime signals on FreeBSD (yet?) or OSX and it seems that the signal used for this purpose doesn't necessarily have to be a realtime one. So such solution would be more universal and also explicit about whether to ignore the signal or not, which seems useful. I could also imagine apps that actually use the SIGRTMAX and host coreclr. Again, the explicit selection of the signal would be helpful in such case.